src/core/algorithms/dc/verifier/dc_verifier.cpp

```diff
@@ -0,0 +1,523 @@
+        }
+    }
+
+    util::Rect<point> box(lower_bound, upper_bound);
+    util::Rect<point> inv_box(lower_bound_inv, upper_bound_inv);
+
+    box.SetBoundType(lower_bound_type, upper_bound_type);
+    inv_box.SetBoundType(lower_bound_type_inv, upper_bound_type_inv);
+
+    return {box, inv_box};
+}
+
+DC DCVerifier::ConvertEqualities(const DC& dc) {
+    std::vector<Predicate> res, preds = dc.GetPredicates();
+    for (auto const& pred : preds) {
+        auto left = pred.GetLeftOperand();
+        auto right = pred.GetRightOperand();
+        if (pred.IsCrossColumn() and pred.IsCrossTuple() and
+            pred.GetOperator().GetType() == OperatorType::kEqual) {
+            auto left_op = ColumnOperand(left.GetColumn(), left.GetTuple());
+            auto right_op = ColumnOperand(right.GetColumn(), right.GetTuple());
+
+            auto less_equal = Operator(OperatorType::kLessEqual);
+            auto greater_equal = Operator(OperatorType::kGreaterEqual);
+
+            res.push_back({less_equal, left_op, right_op});
+            res.push_back({greater_equal, left_op, right_op});
+        } else {
+            res.push_back(pred);
+        }
+    }
+
+    return res;
+}
+
+std::vector<DC> DCVerifier::ConvertDisequalities(const DC& dc) {
+    std::vector<DC> res;
+    std::vector<Predicate> no_diseq_preds, diseq_preds, predicates = dc.GetPredicates();
+
+    for (auto& pred : predicates) {
+        auto op = pred.GetOperator();
+        if (op.GetType() == OperatorType::kUnequal and pred.IsCrossTuple())
+            diseq_preds.push_back(std::move(pred));
+        else
+            no_diseq_preds.push_back(std::move(pred));
+    }
+
+    std::vector<std::vector<Predicate>> temp_preds{std::move(no_diseq_preds)};
+    for (auto const& pred : diseq_preds) {
+        size_t size = temp_preds.size();
+        for (size_t i = 0; i < size; i++) {
+            temp_preds.push_back(temp_preds[i]);
+        }
+
+        ColumnOperand left = pred.GetLeftOperand();
+        ColumnOperand right = pred.GetRightOperand();
+        auto less_pred = Predicate(Operator(OperatorType::kLess), left, right);
+        auto greater_pred = Predicate(Operator(OperatorType::kGreater), left, right);
+
+        for (size_t i = 0; i < temp_preds.size() / 2; i++) {
+            temp_preds[i].push_back(less_pred);
+        }
+
+        for (size_t i = temp_preds.size() / 2; i < temp_preds.size(); i++) {
+            temp_preds[i].push_back(greater_pred);
+        }
+    }
+
+    for (auto& vec_pred : temp_preds) {
+        res.push_back({std::make_move_iterator(vec_pred.begin()),
+                       std::make_move_iterator(vec_pred.end())});
+    }
+
+    return res;
+}
+
+// FIX: DC may contain braces, numbers or spaces, naive separation doesn't belong here.
+std::vector<Predicate> DCVerifier::SplitDC(std::string const& dc_string) {
+    size_t ind, start = 0;
+    std::string token, sep = " and ";
+    std::vector<Predicate> predicates;
+    while (true) {
```

Comment:
Maybe it's better to use
```suggestion
    std::vector<std::string> tokens;
    boost::split(tokens, dc_string, boost::is_any_of(" and "));
    for (auto& token : tokens) {
```

Reply:
Unfortunately, it splits the string be each of the specified characters, not by a whole word

Reply:
Unfortunately I haven't found any intelligible way to split the string